WorkflowUI/Tests/AdaptedEnvironmentScreenTests.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  AdaptedEnvironmentTests.swift
+//  BlueprintUI-Unit-Tests
+//
+//  Created by Kyle Van Essen on 6/28/21.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+import XCTest
+@testable import WorkflowUI
+
+class AdaptedEnvironmentScreenTests: XCTestCase {
+    func test_wrapping() {
+        var environment: ViewEnvironment = .empty
+
+        let screen = TestScreen { environment = $0 }
+            .adaptedEnvironment(key: TestingKey1.self, value: "adapted1.1")
+            .adaptedEnvironment(key: TestingKey1.self, value: "adapted1.2")
+            .adaptedEnvironment(key: TestingKey2.self, value: "adapted2.1")
+            .adaptedEnvironment(key: TestingKey1.self, value: "adapted1.3")
+            .adaptedEnvironment(key: TestingKey2.self, value: "adapted2.2")
+
+        _ = screen.viewControllerDescription(environment: .empty)
+
+        // The inner-most change; the one closest to the screen; should be the value we get.
```

Comment:
at first glance, this seems a bit confusing, and almost feels backwards. i can see why it is true and expected (the adapters are applied from 'outside in'), but it makes the example code above seem more confusing. since the outer wrappers that attempt to adapt earlier keys effectively do nothing, does it make sense to allow them to be written? should there be some form of warning produced if clients attempt to do this? as implemented here, this might not be possible to detect, but just something maybe worth considering.

Reply:
I mentioned this above; but this is basically a somewhat artificial test / a test designed to simulate where these adaptations happen throughout the screen tree; to make sure the "last write wins" is true.

> should there be some form of warning produced if clients attempt to do this

I don't think this is really possible; without _also_ including the `ViewEnvironmentKey` in the generic of the screen; which then also means we'd need to get rid of the keypath and closure-based modifiers.